runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -122,41 +144,73 @@ MM_IndexableObjectAllocationModel::initializeIndexableObject(MM_EnvironmentBase
 	GC_ArrayObjectModel *indexableObjectModel = &extensions->indexableObjectModel;
 	J9IndexableObject *spine = (J9IndexableObject*)initializeJavaObject(env, allocatedBytes);
 	_allocateDescription.setSpine(spine);
+	bool isArrayletDataAdjacentToHeader = false;
+	bool isAllIndexableDataContiguousEnabled = indexableObjectModel->isVirtualLargeObjectHeapEnabled();
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+	isAllIndexableDataContiguousEnabled = isAllIndexableDataContiguousEnabled || indexableObjectModel->isDoubleMappingEnabled();
+#endif /* defined (J9VM_GC_ENABLE_DOUBLE_MAP) */
+
 	if (NULL != spine) {
 		/* Set the array size */
 		if (getAllocateDescription()->isChunkedArray()) {
```

Comment:
think if asserting !offheap is valid

Reply:
if I am reading code correctly, 0-length arrays going here. So for off heap an assertion should be !offheap ||  (0 == _numberOfIndexedFields)